packages/core/test/history.test.ts

```diff
@@ -354,7 +352,9 @@ describe('internal transition with history', () => {
     }
   });
 
-  it('should transition internally to the most recently visited state', () => {
+  // History is not set before the transition is taken, so this does not
```

Comment:
im not entirely sure about this:
1. microstep contains has `exitStates` as its first step: https://www.w3.org/TR/scxml/#microstepProcedure
2. one of the first things `exitStates` does is computing the exit set
3. even internal transitions exit current state according to the second example here: https://www.w3.org/TR/scxml/#N101D7 . The just don't exit its source state, but the source state in here is `second`, not `second.other`
4. so given the fact that `second.other` should be exited here then it should be reentered when this transition targets `.hist` (while it doesnt do it right now and hence fail the test)

Arguably the test description doesnt really tell this story well, so maybe you have thought that it was supposed to test something different? Or maybe it was supposed something different and just was testing the described scenario accidentally? Or maybe I have misinterpreted SCXML semantics?

Reply:
You're right; this will be resolved in a follow-up PR for microsteps